backend/control/control.go

```diff
@@ -169,14 +151,127 @@ func TriggerScan(postgresDb *gorm.DB) error {
 		return err
 	}
 	if sr.Status == "FAILED" {
-		return fmt.Errorf("starting cartography job failed")
+		return fmt.Errorf("scan status: failed")
+	}
+	return nil
+}
+
+// Attempts to run a loop which fetches the account which is ready to be scanned
+// and starts cartography job and execute rules for that account
+func CartographyExecutionLoop(postgresDb *gorm.DB, driver neo4j.Driver) {
```

Comment:
same suggestion for CartographyExecutionLoop as ExecuteRulesLoop with the defer of the Unlock and separate of function inside for loop after sleep